cli/tsc/dts/lib.deno.unstable.d.ts

```diff
@@ -701,20 +701,6 @@ declare namespace Deno {
     close(): void;
   }
 
-  /**
-   *  This magic code used to implement better type hints for {@linkcode Deno.dlopen}
-   *
-   *  @category FFI
-   */
-  type Cast<A, B> = A extends B ? A : B;
-  /** @category FFI */
-  type Const<T> = Cast<
-    T,
-    | (T extends string | number | bigint | boolean ? T : never)
-    | { [K in keyof T]: Const<T[K]> }
-    | []
-  >;
```

Comment:
This was causing an error that looked like the following:

```
Type of property 'struct' circularly references itself in mapped type '{ [K in keyof NativeStructType]: Cast<NativeStructType[K], [] | (NativeStructType[K] extends string | number | bigint | boolean ? NativeStructType[K] : never) | { [K in keyof NativeStructType[K]]: Cast<...>; }>; }'.
```

See this simplified reproduction in [the playground](https://www.typescriptlang.org/play?ts=5.4.3#code/CYUwxgNghgTiAEA7KBbEBnADlMCAiIiA9vAN4BQ88ALgJ6YIByU1AlgG4iMCuKARiBgAVeggC88AETcAzACZJAbko1R8Zm04AFIq0TVBIhvAmTMu-YKUq6xjRxABlajG5hqR8WXhwowIogQtPDoLm7UAFw+IH4BQeosDp4A2gC68AC+yiogAB7mMNSqdomcniYqVAA+CZpcvALCopXwNfbaFgZNDC1tpU5h7p7ZVHkFRXpdAGY4CABiRHCsAOaIc9yI7qwBADwtWrCoIF3o8HkGiMCnvv6Bwe0gKekSN3H3-U8qAHxkLdgwRxOUQOALQJ2UVAy5By+UWRUgUHQpwAqoh0FApiA5ogdJNBDtsWdchcrvAFktVutNmwAj8KFQqKApnpWDTEFFsSMGWAAqFXO5FgAKJkstlRADCvOoBMQXwAlBDMtCqLYEOLEdKAIIAGngACEfhJNUSSac9fAAPzwY1RPWK1XwSVo6VCQ2OjV7BnwITa3rwQVCE2EUl8vTLVpIBqCCN8FaTGNEIgQGKIS3e+BRRAgTgwOV+0jwZIAaXgengAGsQLQiFNvakJVKdkJi6kflCvTU0t9lFDyDznfApohMC4TEgQAB3eAEYgAOlR6Mx2NxlhggpU9IZ-0BgnQUWS0nkkl1Zk6VlSvsh8ER8H7oV9CvIQA).

I can't understand the point of this `Const` code, there's zero tests that fail when I remove it, and it's an unstable feature, so I just removed it.

Reply:
Oh, ok figured it out. This code was emulating `const` type parameters, which is a new language feature as of TS 5.0, so switching to that resolves this issue.